digiroad2-oracle/src/test/scala/fi/liikennevirasto/digiroad2/service/pointasset/TrafficSignServiceSpec.scala

```diff
@@ -334,4 +336,35 @@ class TrafficSignServiceSpec extends FunSuite with Matchers with BeforeAndAfter
       asset.propertyData.find(p => p.publicId == "trafficSigns_info").get.values.head.propertyValue should be ("Additional Info for test")
     }
   }
+
+  test("Pedestrian crossings are filtered") {
+    runWithRollback {
+     when(mockRoadLinkService.getRoadLinksWithComplementaryAndChangesFromVVH(any[BoundingRectangle], any[Set[Int]], any[Boolean])).thenReturn((List(), Nil))
+
+     val roadLink = RoadLink(388553075, Seq(Point(0.0, 0.0), Point(0.0, 20.0)), 10, Municipality, 1, TrafficDirection.BothDirections, Motorway, None, None, Map("MUNICIPALITYCODE" -> BigInt(235)))
+     val adjacentRoadLink = RoadLink(388553074, Seq(Point(0.0, 20.0), Point(0.0, 40.0)), 10, Municipality, 1, TrafficDirection.BothDirections, Motorway, None, None, Map("MUNICIPALITYCODE" -> BigInt(235)))
+
+     val properties = Set(
+        SimpleProperty("trafficSigns_type", List(PropertyValue("7"))),
+        SimpleProperty("trafficSigns_value", List(PropertyValue(""))),
+        SimpleProperty("trafficSigns_info", List(PropertyValue("Pedestrian crossing for test purpose"))))
+
+      service.create(IncomingTrafficSign(0.0, 20.0, 388553075, properties, 1, None), testUser.username, roadLink)
+      service.create(IncomingTrafficSign(0.0, 20.0, 388553075, properties, 1, None), batchProcessName, roadLink)
+      service.create(IncomingTrafficSign(0.0, 20.0, 388553075, properties, 2, None), batchProcessName, roadLink)
+      service.create(IncomingTrafficSign(0.0, 20.0, 388553075, properties, 3, None), batchProcessName, roadLink)
+      service.create(IncomingTrafficSign(0.0, 19.0, 388553075, properties, 1, None), batchProcessName, roadLink)
+
+      service.create(IncomingTrafficSign(0.0, 21.0, 388553074, properties, 1, None), batchProcessName, adjacentRoadLink)
+      service.create(IncomingTrafficSign(0.0, 21.0, 388553074, properties, 2, None), batchProcessName, adjacentRoadLink)
+      service.create(IncomingTrafficSign(0.0, 21.0, 388553074, properties, 3, None), batchProcessName, adjacentRoadLink)
+      service.create(IncomingTrafficSign(0.0, 21.0, 388553074, properties, 1, None), batchProcessName, adjacentRoadLink)
+
+      val result = service.getByBoundingBox(testUser, BoundingRectangle(Point(0.0, 0.0), Point(0.0, 40.0)))
+      val pedestrianCrossings = result.filter(_.propertyData.find(_.publicId == typePublicId).get.values.head.propertyValue.toInt == TrafficSignType.PedestrianCrossing.value)
```

Comment:
Probably we should test if the count is done according with the validity direction and the created by, like: 

```
 val result = service.getByBoundingBox(testUser, BoundingRectangle(Point(0.0, 0.0), Point(0.0, 40.0)))
 val assetsOnLinkId1_vd1 = result.toList.groupBy(_.linkId).find(_._1 == linkId1).map(_._2).getOrElse(Seq.empty[PersistedTrafficSign]).filter(asset => asset.validityDirection == 1 && asset.createdBy.contains("batch_process_trafficSigns"))
 val assetsOnLinkId2_vd1 = result.toList.groupBy(_.linkId).find(_._1 == linkId2).map(_._2).getOrElse(Seq.empty[PersistedTrafficSign]).filter(asset => asset.validityDirection == 1 && asset.createdBy.contains("batch_process_trafficSigns"))
val assetsOnLinkId2_vd2 = result.toList.groupBy(_.linkId).find(_._1 == linkId2).map(_._2).getOrElse(Seq.empty[PersistedTrafficSign]).filter(asset => asset.validityDirection == 2 && asset.createdBy.contains("batch_process_trafficSigns"))

 assetsOnLinkId1_vd1.head.propertyData.filter(_.id == 0).head.values.head.propertyValue.toInt should be (2)
 assetsOnLinkId2_vd1.head.propertyData.filter(_.id == 0).head.values.head.propertyValue.toInt should be (2)
 assetsOnLinkId2_vd2.head.propertyData.filter(_.id == 0).head.values.head.propertyValue.toInt should be (1)
```

because this test is just checking if the bounding box works

Reply:
The filtering is in this case kind of unnecessary. Only a safe way to only show the pedestrian crossings in case other signs have been created on the same spot and the rollback was not successful.